src/python/ensembl/io/genomio/fasta/process.py

```diff
@@ -70,10 +70,9 @@ def prep_fasta_data(
     file_path = Path(fasta_infile)
 
     seqr_to_exclude = set(exclude_seq_regions)
-    if peptide_mode:
-        if genbank_infile is not None:
-            genbank_path = Path(genbank_infile)
-            to_exclude = get_peptides_to_exclude(genbank_path, seqr_to_exclude)
+    if peptide_mode and genbank_infile is not None:
+        genbank_path = Path(genbank_infile)
+        to_exclude = get_peptides_to_exclude(genbank_path, seqr_to_exclude)
```

Comment:
Ok Im happy to revert this back in that case. I thought this was equivalent sorry. It made the change as it was causing issues with the exception raise

Reply:
I believe this behaviour makes more sense and the previous behaviour was a mistake, but may need someone with better understanding of the code to give the final approval. I would leave your suggestion until that time comes.

Reply:
@MatBarba Would you be able to comment on this change. I believed this logic was doing as before the change, but had allowed test to catch raise exception without complaining. Since the PR was made now updated with Jorges suggestion re: ContextManager. So perhaps just revert this change altogether